app/process/watcher/crypto-transfer/crypto-transfer.go

```diff
@@ -157,32 +158,42 @@ func (ctw CryptoTransferWatcher) beginWatching(q *queue.Queue) {
 func (ctw CryptoTransferWatcher) processTransaction(tx transaction.HederaTransaction, q *queue.Queue) {
 	ctw.logger.Infof("New Transaction with ID: [%s]", tx.TransactionID)
 
-	var amount int64
+	var amount string
 	for _, tr := range tx.Transfers {
 		if tr.Account == ctw.accountID.String() {
-			amount = tr.Amount
+			amount = fmt.Sprint(tr.Amount)
 			break
 		}
 	}
 
+	wholeMemoCheck := regexp.MustCompile("^0x([A-Fa-f0-9]){40}-[1-9][0-9]+-[1-9][0-9]+$")
+
 	decodedMemo, e := base64.StdEncoding.DecodeString(tx.MemoBase64)
-	if e != nil || len(decodedMemo) < 42 {
+	if e != nil || len(decodedMemo) < 42 || !wholeMemoCheck.MatchString(string(decodedMemo)) {
 		ctw.logger.Errorf("Could not verify transaction memo - Error: [%s]", e)
```

Comment:
This error is misleading. We are not `verifying` the transaction memo, but `parsing` it